File: src/array/lc11.java
package array;

public class lc11 {
	public static int maxArea(int[] height) {
		int length = height.length;
		int i = 0, j = length - 1;
		int res = 0;
		int area = 0;
		while (i < j) {
			area = (j - i) * Math.min(height[i], height[j]);
			res = Math.max(area, res);
			if (height[i] < height[j])
				++i;
			else {
				--j;
			}
		}
		return res;
	}

	public static void main(String[] args) {
		int[] test = { 1, 8, 6, 2, 5, 4, 8, 3, 7 };
		System.out.println(maxArea(test));
	}
}

File: c++/greedy/lc11.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int maxArea(vector<int>& height) {
		int i=0,j=height.size()-1;
		int res = 0,h=0,x=0;
		while(i<j){
			x=j-i;
			h=min(height[i],height[j]);
			res = max(x*h, res);
			if(height[i]<height[j])i++;
			else j--;
		}
		
        return res;
    }
};
